File: Leetcode/two_pointers/209_MinimumSizeSubarraySum/Solution.java
class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        if(nums.length==0)
            return 0;
        int res = Integer.MAX_VALUE, end = -1, sum = 0;
        for(int start = 0; start<nums.length-1;){
            if( sum<s && end<nums.length-1){
                sum += nums[++end];
            }else{
                sum -= nums[start++];
            }
            if(sum>=s)
                res = Math.min(res, end-start+1);
        }
        return (res!=Integer.MAX_VALUE)? res: 0;
    }
}

File: Leetcode/two_pointers/209_MinimumSizeSubarraySum/Solution.py
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        res, start, end, total = sys.maxsize, 0, -1, 0
        while start < len(nums):
            if total < s and end < len(nums)-1:
                end += 1
                total += nums[end]
            else:
                total -= nums[start]
                start += 1
            if total >= s:
                res = min(res, end-start+1)
        return 0 if res == sys.maxsize else res
